### DISCUSSION TITLE
Controllable animation in 11.X

### DISCUSSION THREAD

**Discussion by Mopy18** [AvaloniaUI/Avalonia#14681] [Q&A]
Hi,

I just upgraded from 0.10 to 11.0.9 and I noticed that we can no longer inherit from IClock and ClockBase. How are we supposed to control an animation from code behind now? The default RunAsync method is very limited because it doesn't allow to start at a specific time, rewind, etc.

I took a look at TopLevel.RequestAnimationFrame but I don't think it's going to work in my case. Basically my app is a sheet music writer/reader, with animations for the player cursor and the notes of an instrument. It looks like this:

https://github.com/AvaloniaUI/Avalonia/assets/110084229/6b8eb79b-3fbb-46c0-8579-20bdf6e18cf4

The animations can't be hardcoded because they are created at runtime based on user input, they can be several minutes long and there are dozens of them that must play in sync and be controllable.

It used to work perfectly with custom Clock, but in 11.X it feels like animations have become a lot more rigid and I don't know how to achieve the same result, apart from rewriting a whole animation system on my own. 

I hope I am missing something and you can help me on that.

**Comment by Gillibald** [comment]
@jmacato

**Comment by Mopy18** [comment]
Hey, any news on this subject, or suggestions to work around the problem?
